clusterDeployer.py

```diff
@@ -83,6 +83,15 @@ def setup_vm(h: host.Host, cfg: NodeConfig, iso_or_image_path: str) -> host.Resu
     name = cfg.name
     mac = cfg.mac
     disk_size_gb = cfg.disk_size
+    cmd = "virsh list --all"
+    ret = h.run(cmd)
+    if f"{name}" in ret.out:
+        logger.info(f"Destroying/Undefining {name} as already existed")
+        cmd = f"virsh destroy {name}"
+        ret = h.run(cmd)
+        cmd = f"virsh undefine {name}"
+        ret = h.run(cmd)
```

Comment:
How come teardown doesn't take care of this?

Reply:
We need to make teardown better then instead instead of duplicating the functionality here. We're calling teardown just before deploying masters. We need to refactor teardown and reuse it from the same place in the code.